### DISCUSSION TITLE
ABC

### DISCUSSION THREAD

**Discussion by GenB31415** [zwicker-group/py-pde#87] [Q&A]
It will be great if in the types of the boundary conditions (BC) will be yet the choice "Absorbing boundary conditions (ABC)". The well-known  technique ABC is used in many problems (with PDE) where the infinite (without boundaries) system  is modelled to suppress the unphysical (numerical) reflections. Thank you.

**Comment by david-zwicker** [comment]
This sounds interesting! What is the mathematical definition of these conditions?

**Reply by GenB31415** [reply]
There are a few papers
1.
Absorbing boundary conditions for reaction–diffusion equations
Jérémie Szeftel
IMA Journal of Applied Mathematics, Volume 68, Issue 2, April 2003, Pages 167–184, https://doi.org/10.1093/imamat/68.2.167
Published: 01 April 2003

Abstract
We treat here of the question of absorbing boundary conditions for nonlinear diffusion equations. 
We use the conditions designed for the linear equation, we prove them to be well posed for the nonlinear problem, 
and through numerical experiments that they are well suited for reaction–diffusion equations.

2.
Analysis of High-Order Absorbing Boundary Conditions for the Schrödinger Equation
Jiwei Zhang,Zhizhong Sun,Xiaonan Wu and Desheng Wang

Communications in Computational Physics , Volume 10 , Issue 3 , September 2011 , pp. 742 - 766
DOI: https://doi.org/10.4208/cicp.280610.161110a

Abstract
The paper is concerned with the numerical solution of Schrödinger equations on an unbounded spatial domain. 
High-order absorbing boundary conditions for one-dimensional domain are derived, and the stability of the reduced initial boundary value problem in the computational interval is proved by energy estimate. Then a second order finite difference scheme is proposed, and the convergence of the scheme is established as well. Finally, numerical examples are reported to confirm our error estimates of the numerical methods.

Other references may be found therein.

**Reply by david-zwicker** [reply]
Unfortunately, I don't have the time to read the papers in detail. Can you give the mathematical equations for the boundary conditions for general PDEs? If this is not easy to implement, I suggest that you add this as a feature request to [issues](https://github.com/zwicker-group/py-pde/issues).

**Reply by GenB31415** [reply]
I see. I guess that the main idea of ABC in a brief form (and also small parts of the python code)  may be found in Ref. //physik.uni-graz.at/~pep/Theses/BachelorThesis_Wachter_2017.pdf
